Mathlib/RingTheory/Binomial.lean

```diff
@@ -176,6 +176,35 @@ instance Int.instBinomialRing : BinomialRing ℤ where
 
 end Nat_Int
 
+section Rat_Algebra
+
+variable {R : Type*} [CommRing R] [CharZero R] [IsLeftCancelMulZero R] [Algebra ℚ R]
+
+/-- Inductive definition of the multichoose function. -/
+noncomputable def multichoose: R → ℕ → R
+  | _, 0 => 1
+  | r, k+1 => ((k+1)⁻¹:ℚ)•(r+k)*(multichoose r k)
+
```

Comment:
```suggestion
```

Reply:
Because factorials are invertible in our setting, we can use a definition that doesn't split into cases.